### PR TITLE
MLI-2949 Upgrade vllm to 0.6.1.post2

### PR DESCRIPTION
**Pull request by yunfeng-scale**
# Pull Request Summary

as title

## Test Plan and Usage Guide

validated both vllm and vllm_batch_v2 work, with google/gemma-2-2b-it model

```
docker run \
    --runtime nvidia \
    --shm-size=16gb \
    --gpus '"device=0"' \
    -v $MODEL_PATH:/workspace/model_files:ro \
    -p 5005:5005 \
    --name vllm \
    ${IMAGE} \
    python -m vllm_server --model model_files --tensor-parallel-size 1 --port 5005 --disable-log-requests
```

```
docker run \
    --runtime nvidia \
    --shm-size=16gb \
    --gpus '"device=6,7"' \
    -v $MODEL_PATH:/workspace/model_files:ro \
    -v ${REPO_PATH}/llm-engine/model-engine/model_engine_server/inference/vllm/examples:/workspace/examples \
    -v ${REPO_PATH}/llm-engine/model-engine/model_engine_server/inference/vllm/vllm_batch.py:/workspace/vllm_batch.py \
    -p 5005:5005 \
    -e CONFIG_FILE=/workspace/examples/v2/sample_config_gemma.json \
    -e MODEL_WEIGHTS_FOLDER=/workspace/model_files \
    --name vllm_batch \
    ${IMAGE_BATCH} \
    python vllm_batch.py
```

### CODE REVIEW

File: model-engine/model_engine_server/inference/vllm/Dockerfile.vllm [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # syntax=docker/dockerfile:1
-ARG VLLM_VERSION=0.5.3.post1
+ARG VLLM_VERSION=0.6.0
```
Comment (edgan8) [inline model-engine/model_engine_server/inference/vllm/Dockerfile.vllm:2]: a `.0` release often has more bugs than a later bugfix release, would it be worth trying to upgrade to [v0.6.1.post2](https://github.com/vllm-project/vllm/releases/tag/v0.6.1.post2) instead?

Comment (yunfeng-scale) [reply model-engine/model_engine_server/inference/vllm/Dockerfile.vllm:2]: sur

**Review by seanshi-scale — APPROVED** [review]

File: model-engine/model_engine_server/inference/vllm/Dockerfile.vllm [hunk 2]
```diff
@@ -9,12 +9,6 @@ RUN apt-get update \
     && rm -rf /var/lib/apt/lists/* \
     apt-get clean
 
-# Need to fix flashinfer at 0.0.8 to support gemma models
```
Comment (dmchoiboi) [inline model-engine/model_engine_server/inference/vllm/Dockerfile.vllm:12]: @yunfeng-scale are we sure we don't need this anymore?

Comment (yunfeng-scale) [reply model-engine/model_engine_server/inference/vllm/Dockerfile.vllm:12]: validated with `google/gemma-2-2b-it`

**Pull request merged**